src/Adapter/Order/QueryHandler/GetOrderForViewingHandler.php

```diff
@@ -711,7 +711,7 @@ private function getOrderMessages(Order $order): OrderMessagesForViewing
 
             $messages[] = new OrderMessageForViewing(
                 (int) $orderMessage['id_customer_message'],
-                $orderMessage['message'],
+                (string) $orderMessage['message'],
```

Comment:
Correct me if I'm wrong, but it seems `message` is not supposed to be null, based on the database structure.

https://github.com/PrestaShop/PrestaShop/blob/289b9872802d214f72d72d490c729b213990bae5/install-dev/data/db_structure.sql#L668

Your fix may work, but it is likely hiding another important issue?

Reply:
@Quetzacoalt91 you're right but the SQL query comes from here https://github.com/PrestaShop/PrestaShop/blob/eff832703974bcef13b7ab7ad168a36c7348a48c/src/Adapter/Customer/CustomerDataProvider.php#L97 There are left joins, so I guess even if `cm.message` can never be null in the table itself, the query result can return a null value if the join doesn't match 🤔

Reply:
Fair enough, and no need for further investigation on the Update side.
Thanks